core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelFile.java

```diff
@@ -495,4 +498,16 @@ public CamelModel getCamelModel(){
 
 		return result;
 	}
+
+	private List<AbstractCamelModelElement> findAllGlobalelementsWithId(String nodeId) {
```

Comment:
There is no need for it now. I just extracted the method to have not too much nested if/for
And I think that for now, for the part that we are supporting, the id must be unique between camel elements and non-camel elements